source-code/telemetry/src/implementation/gitOriginCheck.ts

```diff
@@ -0,0 +1,8 @@
+export function originChecked(getOrigin) {
+	if (getOrigin.startsWith("https://" || getOrigin.endsWith(".git"))) {
+		const checkedOrigin = getOrigin
+		return checkedOrigin
+	} else {
+		return "wrong Origin"
+	}
+}
```

Comment:
1. name the function `validateOrigin`
2. place the function in `@inlang/telemetry` 
3. use the function in every app where the origin is used. 

It would probably be even better if you write a function `getOrigin` that is used in every inlang app, ensuring that the origin is always "get" in the same way.